lib/puppet/functions/simplib/params2hash.rb

```diff
@@ -0,0 +1,26 @@
+# Returns a Hash of the parameters of the calling resource
```

Comment:
Might be worthwhile noting this is intended to be used with classes and defines. When used with a puppet language function returns {} because of the scope search.